--- a/vnfs/libexec/wwmkchroot/include-deb
+++ b/vnfs/libexec/wwmkchroot/include-deb
@@ -3,6 +3,9 @@
 DEB_SRCS="root/sources.list"
 DEB_ISOS="root/iso.list"
 APT_CMD="apt-get -c $CHROOTDIR/$APT_CONF -o Dir::Etc::SourceList=/$CHROOTDIR/"
+# Set DEB_NO_INIT_APT_DIR to prevent prepopulation of /etc/apt
+DEB_NO_INIT_APT_DIR=
+
 
 distro_check() {
     if ! [ -x "/usr/bin/dpkg" ] ; then
@@ -57,8 +60,10 @@
         ARCH="i386"
     fi
 
-    mkdir -p $CHROOTDIR/etc
-    cp -rap /etc/apt $CHROOTDIR/etc
+    if [ -z "$DEB_NO_INIT_APT_DIR" ] ;then
+        mkdir -p $CHROOTDIR/etc
+        cp -rap /etc/apt $CHROOTDIR/etc
+    fi
 
     DEB_SRCS_DIRNAME=`dirname $DEB_SRCS`
     mkdir -m 0755 -p $CHROOTDIR/$DEB_SRCS_DIRNAME
@@ -97,21 +102,25 @@
 }
 
 postchroot() {
+    chroot $CHROOTDIR apt-get update -y
+    chroot $CHROOTDIR apt-get install -y $PKGLIST
+
     # copy a more robust sources.list to our chroot
-    cp $CHROOTDIR/$DEB_SRCS $CHROOTDIR/etc/apt/
+    #cp $CHROOTDIR/$DEB_SRCS $CHROOTDIR/etc/apt/
+
 
     # debootstrap created our base chroot and now
     # we install all the required packages into it
-    APT_CMD="${APT_CMD}${SOURCESPATH}"
-    $APT_CMD update
-    $APT_CMD install $PKGLIST
+    #APT_CMD="${APT_CMD}${SOURCESPATH}"
+    #$APT_CMD update
+    #$APT_CMD install $PKGLIST
 
     touch $CHROOTDIR/fastboot
     return 0
 }
 
 finalize() {
-    $APT_CMD clean
+    chroot $CHROOTDIR apt-get clean
     return 0
 }
 
--- a/vnfs/libexec/wwmkchroot/Makefile.am
+++ b/vnfs/libexec/wwmkchroot/Makefile.am
@@ -1,6 +1,6 @@
 wwmkchrootdir = $(libexecdir)/warewulf/wwmkchroot
 
-dist_wwmkchroot_SCRIPTS = centos-5.tmpl centos-6.tmpl centos-7.tmpl include-rhel rhel-generic.tmpl sl-5.tmpl sl-6.tmpl sl-7.tmpl functions include-deb debian-8.tmpl debian7-32.tmpl debian7-64.tmpl golden-system.tmpl include-suse opensuse-13.tmpl sles-11.tmpl sles-12.tmpl include-ubuntu ubuntu-16.04.tmpl
+dist_wwmkchroot_SCRIPTS = centos-5.tmpl centos-6.tmpl centos-7.tmpl include-rhel rhel-generic.tmpl sl-5.tmpl sl-6.tmpl sl-7.tmpl functions include-deb debian-8.tmpl debian7-32.tmpl debian7-64.tmpl debian-9.tmpl golden-system.tmpl include-suse opensuse-13.tmpl sles-11.tmpl sles-12.tmpl include-ubuntu ubuntu-16.04.tmpl
  
 
 MAINTAINERCLEANFILES = Makefile.in
--- /dev/null
+++ b/vnfs/libexec/wwmkchroot/debian-9.tmpl
@@ -0,0 +1,16 @@
+#DESC: A base 64 bit Debian stretch chroot
+
+# The general Deb include has all of the necessary functions, but requires
+# some basic variables specific to each chroot type to be defined.
+. include-deb
+
+ARCH="amd64"
+SUITE="stretch"
+DEB_MIRROR="http://http.us.debian.org/debian/"
+DEB_NO_INIT_APT_DIR=1
+
+# Install additional packages with debootstrap
+PKGLIST="openssh-server openssh-client isc-dhcp-client pciutils strace nfs-common \
+ethtool iproute2 iputils-ping iputils-arping net-tools linux-image-amd64"
+
+# vim:filetype=sh:syntax=sh:expandtab:ts=4:sw=4:
--- a/vnfs/libexec/wwmkchroot/debian-8.tmpl
+++ b/vnfs/libexec/wwmkchroot/debian-8.tmpl
@@ -9,7 +9,7 @@
 DEB_MIRROR="http://http.us.debian.org/debian/"
 
 # Install additional packages with debootstrap
-PKGLIST="openssh-server openssh-client isc-dhcp-client,pciutils strace,nfs-common \
+PKGLIST="openssh-server openssh-client isc-dhcp-client pciutils strace nfs-common \
 ethtool iproute iputils-ping iputils-arping net-tools firmware-bnx2 linux-image-amd64"
 
 
