--- a/vnfs/libexec/wwmkchroot/include-deb
+++ b/vnfs/libexec/wwmkchroot/include-deb
@@ -57,8 +57,10 @@
         ARCH="i386"
     fi
 
-    mkdir -p $CHROOTDIR/etc
-    cp -rap /etc/apt $CHROOTDIR/etc
+    if [ -z "$DEB_NO_INIT_APT_DIR" ] ;then
+        mkdir -p $CHROOTDIR/etc
+        cp -rap /etc/apt $CHROOTDIR/etc
+    fi
 
     DEB_SRCS_DIRNAME=`dirname $DEB_SRCS`
     mkdir -m 0755 -p $CHROOTDIR/$DEB_SRCS_DIRNAME
--- /dev/null
+++ b/vnfs/libexec/wwmkchroot/debian9-64.tmpl
@@ -0,0 +1,16 @@
+#DESC: A base 64 bit Debian stretch chroot
+
+# The general Deb include has all of the necessary functions, but requires
+# some basic variables specific to each chroot type to be defined.
+. include-deb
+
+ARCH="amd64"
+SUITE="stretch"
+DEB_MIRROR="http://http.us.debian.org/debian/"
+DEB_NO_INIT_APT_DIR=1
+
+# Install additional packages with debootstrap
+PKGLIST="openssh-server openssh-client isc-dhcp-client pciutils strace nfs-common \
+ethtool iproute2 iputils-ping iputils-arping net-tools linux-image-amd64"
+
+# vim:filetype=sh:syntax=sh:expandtab:ts=4:sw=4:
