--- a/cluster/etc/wwfirstboot
+++ b/cluster/etc/wwfirstboot
@@ -80,8 +80,8 @@
 #
 ### BEGIN INIT INFO
 # Provides: wwfirstboot
-# Required-Start: 
-# Required-Stop:
+# Required-Start: $remote_fs
+# Required-Stop: $remote_fs
 # Default-Start: 2 3 4 5
 # Default-Stop: 0 1 6
 # Short-Description: Start the firstboot checks
@@ -101,9 +101,9 @@
 . /etc/init.d/functions
 
 SUBSYSFILE="/var/lock/subsys/wwfirstboot";
-CONFIG="/etc/sysconfig/wwfirstboot.conf";
+CONFIG="/etc/wwfirstboot.conf";
 
-wwfbdir="/usr/libexec/warewulf/wwfirstboot";
+wwfbdir="/usr/lib/warewulf/wwfirstboot";
 
 # What "run" are we doing.
 # PRIMARY   == Initial call from chkconfig setup, starting at S05
@@ -236,6 +236,18 @@
     return $RETVAL;
 }
 
+restart()
+{
+	echo "restart unsupported";
+	return 1;
+}
+
+force-reload()
+{
+	echo "force-reload unsupported";
+	return 1;
+}
+
 ret=0;
 while [ "$#" -gt "0" ]; do
     case $1 in
