--- a/ipmi/Makefile.am
+++ b/ipmi/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = lib 3rd_party initramfs
+SUBDIRS = lib
 
 MAINTAINERCLEANFILES = Makefile.in aclocal.m4 configure compile config.* ltmain.sh install-sh missing
 DISTCLEANFILES = 
--- a/provision/Makefile.am
+++ b/provision/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = lib initramfs etc bin cgi-bin share 3rd_party
+SUBDIRS = lib etc bin cgi-bin share
 
 MAINTAINERCLEANFILES = Makefile.in aclocal.m4 configure compile config.* install-sh missing
 DISTCLEANFILES = 
--- a/ipmi/configure.ac
+++ b/ipmi/configure.ac
@@ -68,8 +68,5 @@
    lib/Warewulf/Makefile
    lib/Warewulf/Module/Makefile
    lib/Warewulf/Module/Cli/Makefile
-   3rd_party/Makefile
-   3rd_party/BSD/Makefile
-   initramfs/Makefile
 )
 
--- a/provision/configure.ac
+++ b/provision/configure.ac
@@ -112,9 +112,6 @@
 AC_OUTPUT(
    Makefile
    warewulf-provision.spec
-   3rd_party/Makefile
-   3rd_party/GPL/Makefile
-   3rd_party/BSD/Makefile
    bin/Makefile
    cgi-bin/Makefile
    etc/Makefile
@@ -122,14 +119,6 @@
    etc/filesystem/Makefile
    etc/filesystem/examples/Makefile
    etc/warewulf-httpd.conf
-   initramfs/Makefile
-   initramfs/capabilities/Makefile
-   initramfs/capabilities/provision-adhoc/Makefile
-   initramfs/capabilities/provision-files/Makefile
-   initramfs/capabilities/provision-vnfs/Makefile
-   initramfs/capabilities/provision-selinux/Makefile
-   initramfs/capabilities/setup-filesystems/Makefile
-   initramfs/capabilities/transport-http/Makefile
    share/Makefile
    lib/Makefile
    lib/Warewulf/Makefile
